### DISCUSSION TITLE
How should multi-part recipes be represented?

### DISCUSSION THREAD

**Discussion by jmpage** [schemaorg/schemaorg#3554] [Q&A] [answered]
I have a recipe which has multiple discrete parts, each of which has its own set of ingredients and steps. I would like to represent these using the [Recipe](https://schema.org/Recipe) schema without combining the ingredients from each part into the `recipeIngredient` property and the steps from each part into the `recipeInstructions` property.

The [`recipeInstructions` property](https://schema.org/recipeInstructions) allows the value to be a `CreativeWork`, which `Recipe` inherits from. I would like to use this to represent each part of my recipe as its own sub-recipe and place these under the `recipeInstructions` property.

Here's a json-ld example of what I am proposing (I simplified a recipe for illustrative purposes):

```json
{
  "@context": "https://schema.org",
  "@type": "Recipe",
  "author": "Alice",
  "name": "Example Recipe",
  "recipeInstructions": [
    {
      "@context": "https://schema.org",
      "@type": "Recipe",
      "name": "Shortcrust pastry",
      "recipeIngredient": [
        "1 cup flour",
        "1 cup butter",
        "1 tbs water"
      ],
      "recipeInstructions": [
        "cut butter into flour",
        "add water and work into crust",
        "form into oval on pan"
      ]
    },
    {
      "@context": "https://schema.org",
      "@type": "Recipe",
      "name": "Choux pastry",
      "recipeIngredient": [
        "1 cup flour",
        "1/4 cup butter",
        "1 cup water",
        "3 eggs"
      ],
      "recipeInstructions": [
        "follow the standard stove top process",
        "place pastry on top of shortcrust",
        "bake together at 400 degrees for 40 minutes"
      ]
    }
  ]
}
```

**Question:** While this is valid according to [validator.schema.org](https://validator.schema.org/), I don't know if it aligns with the intent behind the `recipeInstructions` property. Would it be reasonable to represent each part of my recipe as its own sub-recipe and place these under the `recipeInstructions` property? Are there other approaches that I should consider?

**Sidenote:** I understand that Google has [stricter requirements for recipe schemas](https://developers.google.com/search/docs/appearance/structured-data/recipe) for the purposes of making them accessible to their crawlers and that my proposal contravenes their requirements. I am not concerned by this as my intention isn't to use this for SEO. I am using the `Recipe` schema as a standardized representation of a recipe which I can use to populate a template for my personal static recipes catalog.

Thank you in advance for your help!

**Comment by jvandriel** [comment, marked as answer]
I think there's a happy middle ground to be found between what schema.org allows us to express vs what Google would like to see. Given the use case you described I'd probably write something along these lines:
```json
{
  "@context": "https://schema.org",
  "@type": "Recipe",
  "author": "Alice",
  "name": "Example Recipe",
  "recipeInstructions": [{
    "@type": "HowToStep",
    "position": 1,
    "name": "Make Shortcrust pastry",
    "text": "Follow the recipe for Shortcrust pastry.",
    "item": {
      "@type": "Recipe",
      "name": "Shortcrust pastry",
      "recipeIngredient": [
        "1 cup flour",
        "1 cup butter",
        "1 tbs water"
      ],
      "recipeInstructions": [{
        "@type": "HowToStep",
        "position": 1,
        "name": "cut",
        "text": "cut butter into flour"
      }, {
        "@type": "HowToStep",
        "position": 2,
        "name": "add",
        "text": "add water and work into crust"
      }, {
        "@type": "HowToStep",
        "position": 3,
        "name": "form",
        "text": "form into oval on pan"
      }]
    }
  }, {
    "@type": "HowToStep",
    "position": 2,
    "name": "Make Choux pastry",
    "text": "Follow the recipe for Choux pastry.",
    "item": {
      "@type": "Recipe",
      "name": "Choux pastry",
      "recipeIngredient": [
        "1 cup flour",
        "1/4 cup butter",
        "1 cup water",
        "3 eggs"
      ],
      "recipeInstructions": [{
        "@type": "HowToStep",
        "position": 1,
        "name": "follow",
        "text": "follow the standard stove top process"
      }, {
        "@type": "HowToStep",
        "position": 2,
        "name": "place",
        "text": "place pastry on top of shortcrust"
      }, {
        "@type": "HowToStep",
        "position": 3,
        "name": "bake",
        "text": "bake together at 400 degrees for 40 minutes"
      }]
    }
  }]
}
```

**Reply by jmpage** [reply]
Thank you for this solution @jvandriel. I appreciate how it is consistent with the commonly used (Google crawler required) structure of recipe data by including the `Recipes` on the `HowToSteps`. I hadn't considered this approach.
